File: bindings/python/iota_sdk/types/native_token.py
# Copyright 2023 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0

from dataclasses import dataclass, field
from dataclasses_json import config
from iota_sdk.types.common import hex_str_decoder, HexStr, json


@json
@dataclass
class NativeToken:
    """A native token.

    Attributes:
        id: The unique identifier of the native token.
        amount: The amount of native tokens.
    """
    id: HexStr
    amount: int = field(metadata=config(
        encoder=hex,
        decoder=hex_str_decoder,
    ))

File: bindings/nodejs/lib/types/native_token.ts
import { HexEncodedAmount } from '@iota/types';

export class NativeToken {
    /**
     * Identifier of the native token.
     */
    id: string;
    /**
     * Amount of native tokens of the given Token ID.
     */
    amount: HexEncodedAmount;

    constructor(id: string, amount: HexEncodedAmount) {
        this.id = id;
        this.amount = amount;
    }
}
